Security checks for D7 from Drush Audit

steveworley opened the pull request.

steveworley committed:
Security checks for D7 from Drush Audit

steveworley committed:
Merge branch 'master' into security-tasks-drush-audit

seanhamlin commented:
In discussion with @steveworley, this PR still needs more work before it can be merged. Ideally it would also:

* have merged master
* have tests all passing

steveworley committed:
Adds test cases for new checks

steveworley committed:
Merge branch 'master' into security-tasks-drush-audit

steveworley commented:
Woo! @seanhamlin fixed up the test and merged master :) I added some more tests as well for the new checks that I've added.

steveworley commented:
This branch aims to make my Drush Audit tool obsolete so I'll discontinue that project and continue contributing to this one.

seanhamlin commented on composer.json:
```diff
@@ -17,7 +17,9 @@
         "symfony/yaml": "^3.1",
         "symfony/class-loader": "^3.1",
         "doctrine/annotations": "^1.2",
+        "ircmaxell/random-lib": "^1.2",
```
I have deprecated this lib, in favour of the PHP7 backwards lib random_compat. Is there some reason why you need this?

steveworley replied:
Think that might have been a bad merge resolution.

seanhamlin commented on src/Base/PhantomasCaller.php:
```diff
@@ -52,36 +68,44 @@ public function setDrush(DrushCaller $drush) {
+    // export SITE_AUDIT_HTTP_AUTH_USER=[USERNAME]
+    // export SITE_AUDIT_HTTP_AUTH_PASS=[PASSWORD]
+    else if (!empty(getenv('SITE_AUDIT_HTTP_AUTH_USER'))) {
+      $username = getenv('SITE_AUDIT_HTTP_AUTH_USER');
```
s/SITE_AUDIT/DRUTINY/g

seanhamlin commented on src/Check/D7/ViewsPagination.php:
```diff
@@ -3,7 +3,7 @@
 
 use Drutiny\AuditResponse\AuditResponse;
-use Drutiny\Base\Serializer;
+use Drutiny\Helpers\Serializer;
```
I do like this 'Helpers' namespace. I can move a few classes into there.

seanhamlin commented on src/Check/Security/D7/TextFormat.php:
```diff
@@ -0,0 +1,69 @@
+   * {@inheritdoc}
+   */
+  protected function getFormats() {
+    $sql = "SELECT format, name, cache, status, weight FROM filter_format ff;";
```
Use more `{}` around table names.

seanhamlin commented on src/Check/Security/D7/ViewsAccess.php:
```diff
@@ -0,0 +1,70 @@
+    // We assume that permissions will not be set on other types of views and
+    // their permissions will be controlled via the embed method (blocks, panes
+    // etc.).
+    $sql = "SELECT vv.vid, vv.name, vv.description, vv.human_name, vd.id, vd.display_title, vd.display_options, vd.display_plugin
```
If it helps, I have written a `runScript` method in the drush controller, to which you can execute raw PHP in the context of the site. This maybe of use here.

seanhamlin replied:
See the 'MissingModules' check

seanhamlin reviewed (changes requested):
Wow, looking nice @steveworley! A few things need tidying.

seanhamlin commented:
Also, w00t for combining powers!

steveworley committed:
Code review updates

- Updates the views access check so that it leverages drush and Drupal apis for more consistent results.
- Updates the views base class to convert display options to expected output.
- Updates TextFormat check and adds squiggles for DB escaping.
- Updates environment vars from SITE_AUDIT => DRUTINY.

steveworley committed:
Merge branch 'master' into security-tasks-drush-audit

seanhamlin commented on src/Base/PhantomasCaller.php:
```diff
@@ -52,36 +68,44 @@ public function setDrush(DrushCaller $drush) {
     // export DRUTINY_HTTP_AUTH_USER=[USERNAME]
-    // export DRUTINY_HTTP_AUTH_PASS=[PASSWORD].
-    elseif (!empty(getenv('DRUTINY_HTTP_AUTH_USER'))) {
+    // export DRUTINY_HTTP_AUTH_USER=[PASSWORD]
```
bad merge.

seanhamlin commented on src/scripts/ViewsGetAllViews.php:
```diff
@@ -0,0 +1,12 @@
+<?php
+/**
```
If this comment is made with `//` then it will be stripped prior to base64 encoding. The regex to strip multiline comments is a lot harder to do.

seanhamlin commented on tests/src/Unit/Security/ViewsAccessBaseTest.php:
```diff
@@ -0,0 +1,12 @@
+
+
+class ViewsAccessBaseTest {
+
```
empty test?

steveworley committed:
Code review updates

- Update the phantomas merge
- Adds views test coverage

seanhamlin commented on composer.json:
```diff
@@ -18,6 +18,7 @@
         "symfony/class-loader": "^3.1",
         "doctrine/annotations": "^1.2",
         "twig/twig": "~1.0",
+        "guzzlehttp/guzzle": "^6.2",
```
the lock file needs to be committed too.

seanhamlin commented on src/Base/PhantomasCaller.php:
```diff
@@ -52,36 +68,44 @@ public function setDrush(DrushCaller $drush) {
+
     // export DRUTINY_HTTP_AUTH_USER=[USERNAME]
-    // export DRUTINY_HTTP_AUTH_PASS=[PASSWORD].
+    // export DRUTINY_HTTP_AUTH_USER=[PASSWORD]
```
`DRUTINY_HTTP_AUTH_PASS`
